### PR TITLE
Feat/routing

### PR DESCRIPTION
**Pull request by marshnyash**

### CODE REVIEW

File: src/components/header/movie-search-header.tsx [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+  const handleModalVisibility = () => {
+    setIsModalVisible(!isModalVisible);
+  };
+
+  const classes = useStyles();
+
+  const handleSearch = () => {
+    searchMovie(inputValue);
+    history.push({
+      search: `${new URLSearchParams({ search: `${inputValue}` }).toString()}`,
```
Comment (ObiWaanKenobi) [inline src/components/header/movie-search-header.tsx:59]: According to the task you should implement searching with url path params, not query string:
`localhost/search/Search%20Query`
After that `searchQuery` will become available in `match.params` so you'll don't need to split the query string.

File: src/components/header/movie-search-header.tsx [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+  history,
+  location,
+}: Props) => {
+  const [isModalVisible, setIsModalVisible] = useState(false);
+  const [inputValue, setInputValue] = useState(
+    location?.search?.split("=").slice(1).join()
+  );
+
+  // ? render
+  useEffect(() => {
```
Comment (ObiWaanKenobi) [inline src/components/header/movie-search-header.tsx:44]: You don't need it here. Initial movies list loading is already done in movies container.
Get `searchQuery` with `useParams` and pass it to the `fetchMovies`

**Review by ObiWaanKenobi — COMMENTED** [review]
Well done!

File: src/components/header/movie-search-header.tsx [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+      </Modal>
+    </header>
+  );
+};
+
+const mapStateToProps = (state) => ({
+  search: state.movies.search,
+});
+
+const mapDispatchToProps = (dispatch) => ({
```
[the remark this answers is not in the document]
Comment (marshnyash) [reply src/components/header/movie-search-header.tsx:103]: Could you help me, please?
I'm not sure that I understand correctly.
Thanks in advance.

**Pull request merged**